fix(@desktop/chat): sign and send appears for both recipient and sender when sharing the address

saledjenic opened the pull request:
The reason for this issue is a message where recipient accepted to share his address with sender.
In that message recipient's public key is set as a "from" property of a "Message" object and we
cannot determine which of two users has initiated transaction actually.

This is fixed checking if the "from" address from the "commandParameters" object of the "Message"
is contained as an address in the wallet of logged in user. If yes, means that currently logged in
user has initiated a transaction (he is a sender), otherwise currently logged in user is a
recipient.

We were just sending a transaction, without notifying message about that. Now we call
callPrivateRPC("acceptRequestTransaction".prefix, %* [transactionHash, messageId, signature])
and that notifies message about the change, but only on the sender side. Appropriate message
on the recipient side was not notified about the change. That need to be checked.

Fixes: #2719

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 7d1200a2 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/macos/job/PR-2914/1/) | 2021-07-15 05:39:30 | ~9 min | `macos` | [:package:`dmg`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-210715-052936-7d1200-pr2914.dmg) |

jrainville commented on src/status/signals/messages.nim:
```diff
@@ -362,7 +369,36 @@ proc toMessage*(jsonMsg: JsonNode, pk: string): Message =
+    # the "Message" is contained as an address in the wallet of logged in user. If yes, means that
+    # currently logged in user has initiated a transaction (he is a sender), otherwise currently 
+    # logged in user is a recipient.
+    message.isCurrentUser = currentUserWalletContainsAddress(message.commandParameters.fromAddress)
```
Maybe we could add an `if` to check if there is a `message.commandParameters.fromAddress` first, that way we don't call `currentUserWalletContainsAddress` with an empty sring.

jrainville commented on src/status/signals/messages.nim:
```diff
@@ -362,7 +369,36 @@ proc toMessage*(jsonMsg: JsonNode, pk: string): Message =
+    echo "a"
+    echo "a"
+    echo "a"
+    echo "a"
```
A "little" bit of left over here haha

jrainville commented on ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/TransactionBubble.qml:
```diff
@@ -207,6 +202,44 @@ Item {
+                        transactionSigner.validationError = qsTrId("wrong-password")
+                        return
+                    }
+                    sendingError.text = transactionId
```
I think those errors won't work now that you moved the Connectin in the bubble.

jrainville reviewed (commented):
Looks very good. Nice job.

I just have 2 points to check first

saledjenic commented:
@jrainville I removed that logging code which left over, definitely I overlooked it when I committed changes.

> Maybe we could add an if to check if there is a message.commandParameters.fromAddress first, that way we don't call currentUserWalletContainsAddress with an empty sring.
Actually better place to add that check is currentUserWalletContainsAddress proc, cause that proc may be called later from multiple places and I added it there.

> I think those errors won't work now that you moved the Connectin in the bubble.
It will work everywhere as long as you connect the same valid target (walletModel.transactionsView) to appropriate slot.

jrainville commented:
> It will work everywhere as long as you connect the same valid target (walletModel.transactionsView) to appropriate slot.

Sorry, I think I wasn't clear. What I meant is that the `sendingError` and `transactionSigner` components are probably not accessible since you moved the code to the TransactionBubble. Since the components are in the `SignTransactionModal`, it will probably fail.

jrainville commented:
This probably fixes #2413 as well

saledjenic commented:
> > It will work everywhere as long as you connect the same valid target (walletModel.transactionsView) to appropriate slot.
> 
> Sorry, I think I wasn't clear. What I meant is that the `sendingError` and `transactionSigner` components are probably not accessible since you moved the code to the TransactionBubble. Since the components are in the `SignTransactionModal`, it will probably fail.

Yes, you're right. But another important thing I overlooked is that SignTransactionModal component is used not only from TransactionBubble, but also BrowserLayout and AcceptTransaction and moving connection from the place where it was initially corrupts behaviour in those two components. So I moved it back and updated accordingly.
